src/audio/mic_privacy_manager/mic_privacy_manager_intel.c

```diff
@@ -0,0 +1,236 @@
+		       NOTIFIER_TARGET_CORE_ALL_MASK, settings,
+		       sizeof(struct mic_privacy_settings));
+}
+
+uint32_t mic_privacy_get_dma_zeroing_wait_time(void)
+{
+	return (uint32_t)mic_privacy_api->get_dma_data_zeroing_wait_time();
+}
+
+uint32_t mic_privacy_get_privacy_mask(void)
+{
+	if (mic_privacy_policy == MIC_PRIVACY_HW_MANAGED)
+		return mic_privacy_api->get_dma_data_zeroing_link_select();
+
+	/* hardcoded for FW_MANAGED */
+	return 0xFFFFFFFF;
+}
+
+void mic_privacy_fill_settings(struct mic_privacy_settings *settings, uint32_t mic_disable_status)
+{
+	if (!settings) {
+		LOG_ERR("Invalid mic_privacy_settings pointer");
+		return;
+	}
+
+	settings->mic_privacy_mode = mic_privacy_policy;
+	settings->mic_privacy_state = mic_disable_status;
+	settings->privacy_mask_bits.value = mic_privacy_get_privacy_mask();
+	settings->max_ramp_time = mic_privacy_get_dma_zeroing_wait_time();
+
+	LOG_DBG("mic_privacy_mode = %d, mic_disable_status = %d, \
+		privacy_mask = %x, max_ramp_time_in_ms = %d",
+		settings->mic_privacy_mode,
+		settings->mic_privacy_state,
+		settings->privacy_mask_bits.value,
+		settings->max_ramp_time);
+}
+
+void mic_privacy_set_gtw_mic_state(struct mic_privacy_data *mic_priv_data,
+				   uint32_t mic_disable_status)
+{
+	if (mic_privacy_policy == MIC_PRIVACY_HW_MANAGED) {
+		if (mic_disable_status != 0)
+			mic_priv_data->mic_privacy_state = MIC_PRIV_MUTED;
+		else
+			mic_priv_data->mic_privacy_state = MIC_PRIV_UNMUTED;
+	} else if (mic_privacy_policy == MIC_PRIVACY_FW_MANAGED) {
+		if (mic_disable_status != 0) {
+			LOG_DBG("MUTED");
+			mic_priv_data->mic_privacy_state = MIC_PRIV_MUTED;
+			mic_priv_data->dma_data_zeroing = true;
+			mic_privacy_api->set_fw_mic_disable_status(true);
+		} else {
+			LOG_DBG("UNMUTED");
+			mic_priv_data->mic_privacy_state = MIC_PRIV_UNMUTED;
+			mic_priv_data->dma_data_zeroing = false;
+			mic_privacy_api->set_fw_mic_disable_status(false);
+		}
+	}
+}
+
+void mic_privacy_update_gtw_mic_state(struct mic_privacy_data *mic_priv_data,
+				      uint32_t hw_mic_disable_status)
+{
+	switch (mic_privacy_policy) {
+	case MIC_PRIVACY_HW_MANAGED:
+		mic_privacy_set_gtw_mic_state(mic_priv_data, hw_mic_disable_status);
+		break;
+	case MIC_PRIVACY_FW_MANAGED:
+		mic_privacy_set_gtw_mic_state(mic_priv_data,
+					      mic_privacy_api->get_fw_managed_mic_disable_status());
+		break;
+	default:
+		break;
+	}
+}
+
+void mic_privacy_process(struct comp_dev *dev, struct mic_privacy_data *mic_priv,
+			 struct comp_buffer *buffer, uint32_t copy_samples)
+{
+	uint32_t sg_size_in_bytes;
+
+	sg_size_in_bytes = audio_stream_frame_bytes(&buffer->stream);
+	uint32_t one_ms_in_bytes = sg_size_in_bytes * (buffer->stream.runtime_stream_params.rate / 1000);
+	uint32_t copy_bytes = copy_samples * audio_stream_sample_bytes(&buffer->stream);
+
+	switch (mic_priv->mic_privacy_state) {
+	case MIC_PRIV_UNMUTED:
+		break;
+	case MIC_PRIV_MUTED:
+		buffer_zero(buffer);
```

Comment:
is the fall-through intentional? If so, it should be marked as such per `COMPILER_FALLTHROUGH`